Transceiver 13 zrp - 25.4.1  Handle devices that don't support optical channel output power interval

hmod2001 opened the pull request:
MFG date leaf deviation removed - refer to pr 4708 

Problem
The test was attempting to retrieve the output power interval via gnmi.Get() on all devices without checking if the path is supported. This caused the test to block or fail on devices that don't implement this optional OpenConfig path.
Solution

Added a deviation check using deviations.SkipOpticalChannelOutputPowerInterval(dut) before attempting to retrieve the interval value
Falls back to using the default samplingInterval constant when the path is not supported
Only queries the device-specific interval on platforms that support this telemetry path

Changes

Wrapped the gnmi.Get() call for output power interval in a deviation check
Introduced currentSamplingInterval variable to hold either the device-reported or default sampling interval
Used currentSamplingInterval consistently for all output power stream subscriptions (instant, avg, min, max)

[t13zrplogs.txt](https://github.com/user-attachments/files/23481791/t13zrplogs.txt)

Commit:
25.4.1 update

AmrNJ commented on feature/platform/transceiver/tests/zrp_low_power_mode_test/zrp_low_power_mode_test.go:
```diff
@@ -90,11 +90,20 @@ func validateOutputPower(t *testing.T, streams map[string]*samplestream.SampleSt
-	cfgplugins.InterfaceInitialize(t, dut, operationalMode)
-	cfgplugins.InterfaceConfig(t, dut, dut.Port(t, "port1"))
-	cfgplugins.InterfaceConfig(t, dut, dut.Port(t, "port2"))
+	dp1 := dut.Port(t, "port1")
```
Please change the initial block of the code to match with zr_low_power_mode_test.

AmrNJ commented on feature/platform/transceiver/tests/zrp_low_power_mode_test/zrp_low_power_mode_test.go:
```diff
@@ -33,14 +33,14 @@ import (
 
 var (
-	samplingInterval    = 10 * time.Second
 	operationalModeFlag = flag.Int("operational_mode", 5, "vendor-specific operational-mode for the channel")
```
Change the operational mode to 0.

AmrNJ commented on feature/platform/transceiver/tests/zrp_low_power_mode_test/zrp_low_power_mode_test.go:
```diff
@@ -33,14 +33,14 @@ import (
 
 const (
 	intUpdateTime                 = 5 * time.Minute
```
Update intUpdateTime to 12 minutes

AmrNJ commented on feature/platform/transceiver/tests/zrp_low_power_mode_test/zrp_low_power_mode_test.go:
```diff
@@ -122,14 +128,21 @@ func TestLowPowerMode(t *testing.T) {
 			}
+
+			// Conditionally add description stream based on device support
+			if !deviations.SkipTransceiverDescription(dut) {
```
Please add this deviation to Nokia.

Commit:
address comments

hmod2001 commented:
Addressed the comments - please note this pr is replaced by pr 5207 as it includes the gnsi value update from main - the same comments have been addresses in 5207 as well

hmod2001 committed:
Merge branch 'main' into t13-254.1
